ruma-events/src/lib.rs

```diff
@@ -246,6 +247,37 @@ pub struct RedactedSyncUnsigned {
     pub redacted_because: Option<Box<SyncRedactionEvent>>,
 }
 
+impl From<RedactedUnsigned> for RedactedSyncUnsigned {
+    fn from(redacted: RedactedUnsigned) -> Self {
+        if let Some(why) = redacted.redacted_because {
+            let RedactionEvent {
```

Comment:
I think this `if let` would be more readable as a match with the `RedactionEvent` restructuring as part of the first match arm pattern.

Reply:
Is there a clean way to move it out of the box besides `.map(|b| *b)`?

Reply:
No, the "proper" way is box patterns, which are unstable.